File: framework/src/main/java/com/thornbirds/framework/fragment/ComponentFragment.java
package com.thornbirds.framework.fragment;

import android.view.View;

import androidx.fragment.app.Fragment;

/**
 * Basic definition of Component Activity for Android platform.
 *
 * @author YangLi yanglijd@gmail.com
 */
public abstract class ComponentFragment extends Fragment {
    protected final String TAG = getTAG();

    protected abstract String getTAG();

    protected final <V extends View> V $(int id) {
        return (V) getView().findViewById(id);
    }
}

File: framework/src/main/java/com/thornbirds/framework/fragment/ComponentFragment.kt
package com.thornbirds.framework.fragment

import android.view.View
import androidx.annotation.IdRes
import androidx.fragment.app.Fragment

/**
 * Basic definition of Component Activity for Android platform.
 *
 * @author YangLi yanglijd@gmail.com
 */
abstract class ComponentFragment : Fragment() {

    protected abstract val TAG: String

    protected fun <T : View> findViewById(@IdRes id: Int): T {
        return requireView().findViewById(id)
    }
}